src/components/OptionsCard.jsx

```diff
@@ -1,31 +1,50 @@
 import React, { useEffect } from "react";
 import "../styles/styles.css";
 import useFactle from "../hooks/useFactle";
+import Grid from "./Grid";
 
-const OptionsCard = ({ fact }) => {
-  const { currentGuess, clickHandler } = useFactle(fact);
+const OptionsCard = ({ fact: facts }) => {
+  const {
+    currentGuess,
+    guesses,
+    isCorrect,
+    turn,
+    usedKeys,
+    clickHandler,
+    deleteGuess,
+    enterGuessHandler,
+  } = useFactle(facts);
 
   useEffect(() => {
-    window.addEventListener("click", clickHandler);
-
-    return () => window.removeEventListener("click", clickHandler);
-  }, [clickHandler]);
+    console.log(isCorrect, turn, usedKeys);
+  }, [currentGuess, guesses, isCorrect, turn]);
 
   return (
-    <div className="options-container">
-      {fact &&
-        Object.values(fact).map((fact, index) => (
-          <div
-            style={{ minWidth: "50px", minHeight: "50px" }}
-            className="card"
-            key={index}
-          >
-            {fact}
-          </div>
-        ))}
+    <>
+      <Grid currentGuess={currentGuess} guesses={guesses} turn={turn} />
+      <div className="options-container">
+        {facts &&
+          facts.map(({ id, text, correctPosition }) => {
+            const color = usedKeys[text];
+            return (
+              <div
+                className={color}
+                id="card"
+                key={text}
+                onClick={function (event) {
```

Comment:
Your other `onClick` handlers use arrow functions (`() => {}`) - is there a reason this one is declared using the `function` keyword?

Reply:
Oh... nice catch, I wasn't sure how to use 1 `onClick` handler for multiple events so I used it directly as seen from my search. I've made the update now.